lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMMAv2.cpp

```diff
@@ -552,14 +553,17 @@ Value loadArg(ConversionPatternRewriter &rewriter, Location loc, Value tensor,
   int kWidth = encoding.getMMAv2kWidth();
 
   auto warpsPerCTA = mmaLayout.getWarpsPerCTA();
+  auto order = triton::gpu::getOrder(mmaLayout);
   Value warp = udiv(thread, i32_val(32));
   Value lane = urem(thread, i32_val(32));
-  // Note: warps are currently column major in MMA layout
-  Value warpRowIndex = urem(warp, i32_val(warpsPerCTA[0]));
-  Value warpColIndex =
-      urem(udiv(warp, i32_val(warpsPerCTA[0])), i32_val(warpsPerCTA[1]));
-  Value warpM = urem(warpRowIndex, i32_val(shape[0] / 16));
-  Value warpN = urem(warpColIndex, i32_val(shape[1] / 8));
+
+  SmallVector<Value> multiDimWarpId =
+      delinearize(rewriter, loc, warp, warpsPerCTA, order);
+  unsigned lastAxis = order[order.size() - 1];
+  multiDimWarpId[lastAxis] =
```

Comment:
@Jokeren This is why I was asking whether the last dim also needs `% dimSize` in [here](https://github.com/openai/triton/pull/1816#discussion_r1239029353), I wouldn't need to do this outside delinearize.

Reply:
I have cleaned it up. I'll handle it if this PR gets merged before the other.